File: iosApp/Screens/AuthScreen.swift
import SwiftUI
import shared

struct AuthScreen: View {
    @State private var viewModel = KoinHelperSwift.shared.getProfileViewModel()
    @State private var state = ProfileState(user: nil, isAuthorized: false, stats: nil, usernameInput: "", passwordInput: "", isLoading: false, error: nil)
    @State private var isLoginTab = true
    
    @State private var alertMessage: String?
    @State private var showAlert: Bool = false
    
    var body: some View {
        NavigationView {
            ZStack {
                AppColors.background.edgesIgnoringSafeArea(.all)
                
                VStack(spacing: 20) {
                    Spacer().frame(height: 20)

                    ZStack {
                        Circle()
                            .fill(AppColors.primary.opacity(0.1))
                            .frame(width: 80, height: 80)
                        
                        Image(systemName: "film")
                            .resizable()
                            .aspectRatio(contentMode: .fit)
                            .frame(width: 44, height: 44)
                            .foregroundColor(AppColors.primary)
                    }
                    
                    Text("CineVerse")
                        .font(AppTypography.headingLarge)
                        .foregroundColor(AppColors.textPrimary)
                        .fontWeight(.bold)
                    
                    Text("Your Ultimate Movie Companion")
                        .font(AppTypography.bodyMedium)
                        .foregroundColor(AppColors.textSecondary)
                        .padding(.top, -10)
                        .padding(.bottom, 20)

                    HStack(spacing: 4) {
                        Button(action: { isLoginTab = true }) {
                            Text("Log In")
                                .font(.system(size: 16, weight: .bold))
                                .foregroundColor(isLoginTab ? .white : AppColors.textSecondary)
                                .frame(maxWidth: .infinity)
                                .frame(height: 44)
                                .background(isLoginTab ? AppColors.primary : Color.clear)
                                .cornerRadius(8)
                        }
                        
                        Button(action: { isLoginTab = false }) {
                            Text("Register")
                                .font(.system(size: 16, weight: .bold))
                                .foregroundColor(!isLoginTab ? .white : AppColors.textSecondary)
                                .frame(maxWidth: .infinity)
                                .frame(height: 44)
                                .background(!isLoginTab ? AppColors.primary : Color.clear)
                                .cornerRadius(8)
                        }
                    }
                    .padding(4)
                    .background(AppColors.surface)
                    .cornerRadius(12)
                    .padding(.horizontal)
                    
                    Spacer().frame(height: 10)
                    
                    CineVerseTextField(
                        placeholder: "Username",
                        text: Binding(
                            get: { state.usernameInput },
                            set: { viewModel.sendIntent(intent: ProfileIntent.TypeUsername(username: $0)) }
                        ),
                        leadingIcon: "person.fill"
                    )
                    .padding(.horizontal)
                    
                    CineVerseTextField(
                        placeholder: isLoginTab ? "Password" : "Password (min 4 characters)",
                        text: Binding(
                            get: { state.passwordInput },
                            set: { viewModel.sendIntent(intent: ProfileIntent.TypePassword(password: $0)) }
                        ),
                        isSecure: true,
                        leadingIcon: "lock.fill"
                    )
                    .padding(.horizontal)
                    
                    Spacer().frame(height: 10)
                    
                    CineVerseButton(
                        text: isLoginTab ? "Log In" : "Create Account",
                        onClick: {
                            if isLoginTab {
                                viewModel.sendIntent(intent: ProfileIntent.Login())
                            } else {
                                viewModel.sendIntent(intent: ProfileIntent.Register())
                            }
                        },
                        isLoading: state.isLoading
                    )
                    .padding(.horizontal)
                    
                    Spacer()
                }
            }
            .navigationBarHidden(true)
            .alert(isPresented: $showAlert) {
                Alert(title: Text("Auth Status"), message: Text(alertMessage ?? ""), dismissButton: .default(Text("OK")))
            }
        }
        .task {
            for await currentState in viewModel.state {
                if let currentState = currentState {
                    self.state = currentState
                }
            }
        }
        .task {
            for await currentEffect in viewModel.effect {
                switch currentEffect {
                case _ as ProfileEffect.AuthSuccess:
                    break
                case let errorEffect as ProfileEffect.AuthError:
                    self.alertMessage = errorEffect.message
                    self.showAlert = true
                default:
                    break
                }
            }
        }
    }
}

File: androidApp/features/profile/src/main/kotlin/com/cineverse/android/features/profile/AuthScreen.kt
package com.cineverse.android.features.profile

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material.icons.filled.Person
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.SnackbarHostState
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.unit.dp
import com.cineverse.core.designsystem.components.CineVerseButton
import com.cineverse.core.designsystem.components.CineVerseTextField
import com.cineverse.presentation.profile.ProfileEffect
import com.cineverse.presentation.profile.ProfileIntent
import com.cineverse.presentation.profile.ProfileViewModel
import org.koin.compose.koinInject

@Composable
fun AuthScreen(
    snackbarHostState: SnackbarHostState,
    modifier: Modifier = Modifier,
    viewModel: ProfileViewModel = koinInject()
) {
    val state by viewModel.state.collectAsState()
    var isLoginTab by remember { mutableStateOf(true) }

    LaunchedEffect(Unit) {
        viewModel.effect.collect { effect ->
            when (effect) {
                is ProfileEffect.AuthSuccess -> {
                    snackbarHostState.showSnackbar("Welcome to CineVerse!")
                }
                is ProfileEffect.AuthError -> {
                    snackbarHostState.showSnackbar(effect.message)
                }
            }
        }
    }

    Column(
        modifier = modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background) 
            .padding(24.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        
        Box(
            modifier = Modifier
                .size(80.dp)
                .clip(CircleShape)
                .background(MaterialTheme.colorScheme.primary.copy(alpha = 0.1f)),
            contentAlignment = Alignment.Center
        ) {
            Icon(
                imageVector = Icons.Default.PlayArrow,
                contentDescription = "Logo",
                modifier = Modifier.size(44.dp),
                tint = MaterialTheme.colorScheme.primary
            )
        }

        Spacer(modifier = Modifier.height(24.dp))

        Text(
            text = "CineVerse",
            style = MaterialTheme.typography.headlineLarge,
            color = MaterialTheme.colorScheme.onBackground,
            fontWeight = FontWeight.Bold
        )

        Text(
            text = "Your Ultimate Movie Companion",
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
            modifier = Modifier.padding(top = 4.dp, bottom = 32.dp)
        )

        Row(
            modifier = Modifier
                .fillMaxWidth()
                .background(MaterialTheme.colorScheme.surface, RoundedCornerShape(12.dp))
                .padding(4.dp),
            horizontalArrangement = Arrangement.spacedBy(4.dp)
        ) {
            val activeColor = MaterialTheme.colorScheme.primary
            val inactiveColor = Color.Transparent

            Box(
                modifier = Modifier
                    .weight(1f)
                    .clip(RoundedCornerShape(8.dp))
                    .background(if (isLoginTab) activeColor else inactiveColor)
                    .clickable { isLoginTab = true }
                    .padding(vertical = 12.dp),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    text = "Log In",
                    color = if (isLoginTab) MaterialTheme.colorScheme.onPrimary else MaterialTheme.colorScheme.onSurfaceVariant,
                    fontWeight = FontWeight.Bold
                )
            }

            Box(
                modifier = Modifier
                    .weight(1f)
                    .clip(RoundedCornerShape(8.dp))
                    .background(if (!isLoginTab) activeColor else inactiveColor)
                    .clickable { isLoginTab = false }
                    .padding(vertical = 12.dp),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    text = "Register",
                    color = if (!isLoginTab) MaterialTheme.colorScheme.onPrimary else MaterialTheme.colorScheme.onSurfaceVariant,
                    fontWeight = FontWeight.Bold
                )
            }
        }

        Spacer(modifier = Modifier.height(28.dp))

        CineVerseTextField(
            value = state.usernameInput,
            onValueChange = { viewModel.sendIntent(ProfileIntent.TypeUsername(it)) },
            placeholder = "Username",
            modifier = Modifier.fillMaxWidth(),
            leadingIcon = {
                Icon(
                    imageVector = Icons.Default.Person,
                    contentDescription = "Username",
                    tint = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
        )

        Spacer(modifier = Modifier.height(16.dp))

        CineVerseTextField(
            value = state.passwordInput,
            onValueChange = { viewModel.sendIntent(ProfileIntent.TypePassword(it)) },
            placeholder = if (isLoginTab) "Password" else "Password (min 4 characters)",
            modifier = Modifier.fillMaxWidth(),
            visualTransformation = PasswordVisualTransformation(),
            leadingIcon = {
                Icon(
                    imageVector = Icons.Default.Lock,
                    contentDescription = "Password",
                    tint = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
        )

        Spacer(modifier = Modifier.height(32.dp))

        CineVerseButton(
            text = if (isLoginTab) "Log In" else "Create Account",
            onClick = {
                if (isLoginTab) {
                    viewModel.sendIntent(ProfileIntent.Login)
                } else {
                    viewModel.sendIntent(ProfileIntent.Register)
                }
            },
            modifier = Modifier.fillMaxWidth(),
            isLoading = state.isLoading
        )
    }
}
